ui/src/containers/TaskContainer.jsx

```diff
@@ -1,29 +1,42 @@
 import React from 'react';
-import { bindActionCreators } from 'redux';
-import { connect } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import Characterisation from '../components/Tasks/Characterisation';
 import DataCollection from '../components/Tasks/DataCollection';
 import Helical from '../components/Tasks/Helical';
 import Mesh from '../components/Tasks/Mesh';
-import AddSampleContainer from '../components/Tasks/AddSample';
+import AddSample from '../components/Tasks/AddSample';
 import Workflow from '../components/Tasks/Workflow';
 import Interleaved from '../components/Tasks/Interleaved';
 import XRFScan from '../components/Tasks/XRFScan';
 import EnergyScan from '../components/Tasks/EnergyScan';
 import GenericTaskForm from '../components/Tasks/GenericTaskForm';
 
-import { hideTaskParametersForm, showTaskForm } from '../actions/taskForm';
+import { hideTaskParametersForm } from '../actions/taskForm';
 
 import { addTask, updateTask } from '../actions/queue';
 
-class TaskContainer extends React.Component {
-  constructor(props) {
-    super(props);
-    this.addTask = this.addTask.bind(this);
-  }
+function TaskContainer() {
+  const dispatch = useDispatch();
+
+  const sampleList = useSelector((state) => state.sampleGrid.sampleList);
```

Comment:
Same as before, all these will eventually move down into child components once they're all converted to function components.